### PR TITLE
add cd script to deploy to staging env

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What problem does this PR solve? 

This PR adds a jenkins pipeline file to deploy tidb-operator to the staging env

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: ci/deploy_tidb_operator_staging.groovy [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+@Field
+def values = '''
+controllerManager:
+  replicas: 2
+scheduler:
+  replicas: 2
+admissionWebhook:
+  create: true
+  hooksEnabled:
+    statefulSets: true
```
Comment (Yisaer) [inline ci/deploy_tidb_operator_staging.groovy]: StatefulSets webhook is used for canary upgrading. Is it necessary to be enabled in the default?

File: ci/deploy_tidb_operator_staging.groovy [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                deleteDir()
+
+                dir("${WORKSPACE}/operator-staging") {
+                    stage('Download tidb-operator binary') {
+                        GITHASH = sh(returnStdout: true, script: "curl ${UCLOUD_OSS_URL}/refs/pingcap/operator/${BUILD_BRANCH}/centos7/sha1").trim()
+                        sh "curl ${UCLOUD_OSS_URL}/builds/pingcap/operator/${GITHASH}/centos7/tidb-operator.tar.gz | tar xz"
+                    }
+                    stage('Push tidb-operator Docker Image to Harbor') {
+                        docker.withRegistry("https://hub.pingcap.net", "harbor-pingcap") {
+                            sh"""
```
Comment (onlymellb) [inline ci/deploy_tidb_operator_staging.groovy]: Indentation

File: ci/deploy_tidb_operator_staging.groovy [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                        docker build --network=host -t hub.pingcap.net/jenkins/tidb-operator:${GITHASH} images/tidb-operator
+                        docker push hub.pingcap.net/jenkins/tidb-operator:${GITHASH}
+                        """
+                        }
+                    }
+
+                    stage('Deploy tidb-operator to staging') {
+                        ansiColor('xterm') {
+                            writeFile file: 'values.yaml', text: "${values}"
+                            sh """
```
Comment (onlymellb) [inline ci/deploy_tidb_operator_staging.groovy]: ditto

File: ci/deploy_tidb_operator_staging.groovy [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                        # ensure helm
+                        wget https://storage.googleapis.com/kubernetes-helm/helm-v2.9.1-linux-amd64.tar.gz
+                        tar -zxvf helm-v2.9.1-linux-amd64.tar.gz
+                        chmod +x linux-amd64/helm
+
+                        # deploy to staging
+                        export KUBECONFIG=/home/jenkins/.kubeconfig/operator_staging
```
Comment (cofyc) [inline ci/deploy_tidb_operator_staging.groovy]: pre-created manually on this delivery node?

File: ci/deploy_tidb_operator_staging.groovy (hunk 1 above)
Comment (aylei) [reply ci/deploy_tidb_operator_staging.groovy]: IMO, yes, we have a section about canary release in the IDC manual

File: ci/deploy_tidb_operator_staging.groovy (hunk 4 above)
Comment (aylei) [reply ci/deploy_tidb_operator_staging.groovy]: it is a secret that mounted to the delivery Pod

File: ci/deploy_tidb_operator_staging.groovy [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                dir("${WORKSPACE}/operator-staging") {
+                    stage('Download tidb-operator binary') {
+                        GITHASH = sh(returnStdout: true, script: "curl ${UCLOUD_OSS_URL}/refs/pingcap/operator/${BUILD_BRANCH}/centos7/sha1").trim()
+                        sh "curl ${UCLOUD_OSS_URL}/builds/pingcap/operator/${GITHASH}/centos7/tidb-operator.tar.gz | tar xz"
+                    }
+                    stage('Push tidb-operator Docker Image to Harbor') {
+                        docker.withRegistry("https://hub.pingcap.net", "harbor-pingcap") {
+                            sh"""
+                        docker build --network=host -t hub.pingcap.net/jenkins/tidb-operator:${GITHASH} images/tidb-operator
+                        docker push hub.pingcap.net/jenkins/tidb-operator:${GITHASH}
```
Comment (onlymellb) [inline ci/deploy_tidb_operator_staging.groovy]: Refer [here](https://github.com/pingcap/tidb-operator/blob/master/ci/release_tidb_operator_binary_and_image.groovy#L24-L29) to push image?

Comment (aylei) [reply ci/deploy_tidb_operator_staging.groovy]: addressed in https://github.com/pingcap/tidb-operator/pull/1505/commits/8003622b4424e40ab00c6f38ef3c05b1aa49939d

**Pull request merged**